ecs-cli/modules/cli/logs/logs_app.go

```diff
@@ -0,0 +1,274 @@
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package logs
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	cwlogsclient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/cloudwatchlogs"
+	ecsclient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/ecs"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+const (
+	followLogsSleepSeconds = 30
+)
+
+// Logs is the action for logsCommand. It retrieves container logs for a task from CloudWatch
+func Logs(c *cli.Context) {
+	err := validateLogFlags(c)
+	if err != nil {
+		logrus.Fatal("Error executing 'logs': ", err)
+	}
+	rdwr, err := config.NewReadWriter()
+	if err != nil {
+		logrus.Fatal("Error executing 'logs': ", err)
+	}
+	ecsParams, err := config.NewCliParams(c, rdwr)
+	if err != nil {
+		logrus.Fatal("Error executing 'logs': ", err)
+	}
+
+	ecsClient := ecsclient.NewECSClient()
+	ecsClient.Initialize(ecsParams)
+	request, logRegion, err := logsRequest(c, ecsClient, ecsParams)
+	if err != nil {
+		logrus.Fatal("Error executing 'logs': ", err)
+	}
+
+	cwLogsClient := cwlogsclient.NewCloudWatchLogsClient(ecsParams, logRegion)
+
+	printLogEvents(c, request, cwLogsClient)
+
+}
+
+func logsRequest(context *cli.Context, ecsClient ecsclient.ECSClient, params *config.CliParams) (*cloudwatchlogs.FilterLogEventsInput, string, error) {
+	taskID := context.String(command.TaskIDFlag)
+	taskDefNameOrArn := context.String(command.TaskDefinitionFlag)
+
+	if taskDefNameOrArn == "" {
+		tasks, err := getTaskDefArn(context, ecsClient, params)
+		if err != nil {
+			return nil, "", err
+		}
+		taskDefNameOrArn = aws.StringValue(tasks[0].TaskDefinitionArn)
+	}
+
+	taskDef, err := ecsClient.DescribeTaskDefinition(taskDefNameOrArn)
+	if err != nil {
+		return nil, "", errors.Wrap(err, fmt.Sprintf("Failed to Describe TaskDefinition; try using --%s to specify the Task Defintion.", command.TaskDefinitionFlag))
+	}
+
+	var logGroup, logRegion *string
+	var prefixes map[*string]*string
+	if containerName := context.String(command.ContainerNameFlag); containerName != "" {
+		logGroup, logRegion, prefixes, err = getLogConfigurationSingleContainerCase(taskDef, taskID, containerName)
+	} else {
+		logGroup, logRegion, prefixes, err = getLogConfiguration(taskDef, taskID)
+	}
+	if err != nil {
+		return nil, "", errors.Wrap(err, "Failed to get log configuration")
+	}
+
+	streams := logStreams(prefixes, taskID)
+
+	input, err := filterLogEventsInputFromContext(context)
```

Comment:
Hm, since `input` is the output of this method, would it be possible to give it a clearer name?